com.unity.render-pipelines.universal/Runtime/UniversalAdditionalCameraData.cs

```diff
@@ -84,6 +84,109 @@ public static UniversalAdditionalCameraData GetUniversalAdditionalCameraData(thi
 
             return cameraData;
         }
+
+        /// <summary>
+        /// Returns the VolumeFrameworkUpdateMode set on the camera.
+        /// </summary>
+        /// <param name="camera"></param>
+        /// <returns></returns>
+        public static VolumeFrameworkUpdateMode GetVolumeFrameworkUpdateMode(this Camera camera)
+        {
+            UniversalAdditionalCameraData cameraData = camera.GetUniversalAdditionalCameraData();
+            return cameraData.volumeFrameworkUpdateMode;
+        }
+
+        /// <summary>
+        /// Sets the VolumeFrameworkUpdateMode for the camera.
+        /// </summary>
+        /// <param name="camera"></param>
+        /// <param name="mode"></param>
+        public static void SetVolumeFrameworkUpdateMode(this Camera camera, VolumeFrameworkUpdateMode mode)
+        {
+            UniversalAdditionalCameraData cameraData = camera.GetUniversalAdditionalCameraData();
+            cameraData.volumeFrameworkUpdateMode = mode;
+
+            // We only update the local volume stacks for cameras set to ViaScripting.
+            // Otherwise it will be updated in every frame.
+            if (!cameraData.requiresVolumeFrameworkUpdate)
+            {
+                camera.UpdateVolumeStack(cameraData);
+            }
+        }
+
+        /// <summary>
+        /// Updates the volume stack for this camera.
+        /// This function will only update the stack when the camera has VolumeFrameworkUpdateMode set to ViaScripting
+        /// or when it set to UsePipelineSettings and the update mode on the Render Pipeline Asset is set to ViaScripting.
+        /// </summary>
+        /// <param name="camera"></param>
+        public static void UpdateVolumeStack(this Camera camera)
+        {
+            UniversalAdditionalCameraData cameraData = camera.GetUniversalAdditionalCameraData();
+            camera.UpdateVolumeStack(cameraData);
+        }
+
+        /// <summary>
+        /// Updates the volume stack for this camera.
+        /// This function will only update the stack when the camera has ViaScripting selected or if
+        /// the camera is set to UsePipelineSettings and the Render Pipeline Asset is set to ViaScripting.
+        /// </summary>
+        /// <param name="camera"></param>
+        /// <param name="cameraData"></param>
+        public static void UpdateVolumeStack(this Camera camera, UniversalAdditionalCameraData cameraData)
```

Comment:
I'm a little bit confused between `UpdateVolumeStack` and the internal `UpdateVolumeFramework` as they have some duplicated code / logic in them. It seems the `UpdateVolumeFramework` is essentially `UpdateVolumeStack` + reset and should update check?